qiskit/transpiler/passes/mapping/cx_direction.py

```diff
@@ -56,10 +56,8 @@ def run(self, dag):
         new_dag = DAGCircuit()
 
         if self.layout is None:
-            if self.property_set["layout"]:
```

Comment:
removing this seems problematic, since this pass relies on a layout calculated by a  previous pass. here you are overriding this to always be the trivial layout. 

If the issue is with LegacySwap, then that pass should be modified, not CXDirection which is supposed to work with all Swappers. unfortunately we don't have enough integration tests for the transpiler, but my guess is this would break the BasicSwap->CXDirection flow.

Reply:
I thinks this is part of the bigger discussion on which suppose to be the output of a swapper. True, BasicSwap -> CXDirection probably does not work. But that's because BasicSwap and LegacySwap are not having the same output. I think, for the sake a smaller PR, we can attack this in a different one.

Reply:
After talking with you, I will have another look to this issue.